### PR TITLE
feat: Relationship between HPA and other k8s components.

### PR DESCRIPTION
**Pull request by RipulHandoo**
**Notes for Reviewers**
1. Relationship between k8s Horizontal Pod Auto-Scalar and other resources.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by saurabh100ni** [comment]
Hi @RipulHandoo 
Thanks for your PR! I encourage you to share this in the Meshery Development Meeting tomorrow at 8:30 PM IST. Share your screen, and walk us through it. It's cool to take and share feedback. Check the [Doc](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit) here to include this PR.

You're welcome at the meeting

Check to [community calendar](https://layer5.io/community/calendar) for more details of meetings at Layers.

File: server/meshmodel/kubernetes/relationships/hierarchical_inventory.json [hunk 1]
```diff
@@ -247,6 +247,70 @@
...
+    {
+      "allow": {
+        "from": [
+          {
+            "kind": "Deployment",
+            "model": "kubernetes",
+            "patch": {
+              "patchStrategy": "replace",
+              "mutatorRef": [
+                ["apiVersion"],
```
Comment (MUzairS15) [inline server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:260]: The `mutatorRef` path is invalid, willl you ensure the correct ref is present?

Comment (RipulHandoo) [reply server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:260]: After discussion, it's confirmed that the mutatorRef path is correct for current implementation.

Comment (leecalcote) [reply server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:260]: A screenshot, would be great.

**Comment by RipulHandoo** [comment]
@MUzairS15 @leecalcote I have manually filled in the fields for `deployment `and `HPA `in the [Relationship Master Design](https://playground.meshery.io/extension/meshmap?mode=design&design=5f7a16fe-00bb-43e0-b752-05913156535d). Looks good to me. Please take a look!
image

**Review by MUzairS15 — COMMENTED** [review]
@RipulHandoo, Will you drop in a video where you again create existing (network/mount/permission) relationships so that we are covered fully?

You may also want to showcase the working of new relationships defined in this PR.

**Comment by leecalcote** [comment]
This is some pretty cool stuff.

**Comment by RipulHandoo** [comment]
@MUzairS15 In the video, I've added all the relationships for which I raised a PR. However, I manually added the values for now.
https://github.com/meshery/meshery/assets/107461226/42d2800b-d652-43cc-8656-57db0d843e23

**Comment by RipulHandoo** [comment]
@MUzairS15 This PR is ready to merge and it's working as expected.

https://github.com/meshery/meshery/assets/107461226/956c6c26-b429-429f-975b-cb330acc1686

**Review by MUzairS15 — COMMENTED** [review]
No need to manually update the HPA definition.

**Comment by MUzairS15** [comment]
I am talking about the HPA component definition, were you have edited the genealogy.

**Comment by RipulHandoo** [comment]
> I am talking about the HPA component definition, were you have edited the genealogy.
Without that will it work. 
Ok I will make changes in the PR.

**Review by MUzairS15 — APPROVED** [review]
Thanks @RipulHandoo

**Pull request merged**